lib/index.ts

```diff
@@ -186,13 +185,47 @@ async function getCallGraph(
     debug('got call graph successfully');
     return callGraph;
   } catch (e) {
-    const userMessage = formatCallGraphError(e.message);
     debug('call graph error: ' + e);
-    debug(userMessage);
-    const err: CallGraphError = {
-      message: userMessage,
-      innerError: e,
-    };
-    return err;
+    if (e instanceof MissingTargetFolderError) {
```

Comment:
I would move all this logic to java call graph builder, seems more related to it...
I think you can make those user messages part of the error definition there.

Reply:
Probably will be less `if`s, if you you want you can use the `userMessage` attribute

Reply:
I know what you mean. I thought if this should be in here or in the call graph builder. The reason I chose to put it here is because this is just translating the call graph errors--which can be thought of as implementation details--to generic user messages, and propagating the actual error up the chain for analytics. That's why I thought the right place is here, since the call graph builder is not necessarily an end user component, but I could be convinced otherwise.

Reply:
I think that thing that popped up here was that it should be part of the error class, instead of creating many ifs.
One other reason is that once we gonna implement it for Gradle, we would need to copy this logic the snyk-gradle-plugin.